### PR TITLE
Breakout Magnetometers

### PR DESCRIPTION
**Pull request by shihaocao**
# Breakout Magnetometers
The objective of this PR is to be able to interact with MAG1 and MAG2 seperately, and simultaneously query both MAG's over I2C.

Addresses #212 
### Summary of changes
- Update constants.hpp (adcs), many registers/constants were renamed
- Update ADCSSW to support magnetometer breakout
- Updated ADCS Driver for ^
- Updated FSW Control Tasks for ^
- Updated FSW Unit tests for ^
- Updated ADCSCheckoutCase
- Updated flow_data.csv
- Minor QOL /tools/ changes

### Testing
Ptest testing shows that the whole loop is capable of reading from both MAG's simultaneously. This is sufficient for this PR.
Further nuanced ADCS ptesting will come with upgrades to the ADCSCheckoutCase (added to ticket).

### Constants
No real changes.

### Telemetry
Magnetometer related state fields were both added and renamed.
Telemetry was added to `flow_data.csv`

### Documentation Evidence
Added to ReadTheDocs

### CODE REVIEW

File: src/fsw/FCCode/AttitudeEstimator.cpp [hunk 1]
```diff
@@ -43,8 +44,11 @@ void AttitudeEstimator::set_data(){
     const d_vector_t r_ecef = pos_vec_ecef_fp->get();
     data.r_ecef = {r_ecef[0], r_ecef[1], r_ecef[2]};
```
Comment (shihaocao) [inline src/fsw/FCCode/AttitudeEstimator.cpp:46]: @kkrol27 How would you like to handle the section below?

File: src/fsw/FCCode/AttitudeEstimator.cpp [hunk 2]
```diff
@@ -43,8 +44,11 @@ void AttitudeEstimator::set_data(){
     const d_vector_t r_ecef = pos_vec_ecef_fp->get();
     data.r_ecef = {r_ecef[0], r_ecef[1], r_ecef[2]};
 
-    const f_vector_t mag_vec = mag_vec_fp->get();
-    data.b_body = {mag_vec[0], mag_vec[1], mag_vec[2]};
+    const f_vector_t mag1_vec = mag1_vec_fp->get();
```
Comment (shihaocao) [inline src/fsw/FCCode/AttitudeEstimator.cpp:47]: @kkrol27 How'd you like to handle the section below?

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 3]
```diff
@@ -124,7 +129,8 @@ void ADCSBoxMonitor::execute(){
     std::array<float, adcs::ssa::num_sun_sensors> ssa_voltages;
     ssa_voltages.fill(0);
 
-    f_vector_t mag_vec;
+    f_vector_t mag1_vec;
+    f_vector_t mag2_vec;
```
Comment (kkrol27) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp:133]: If you're going through this atm, would you mind switching things over to `lin::Vector3f` state fields instead? I know most of the flight computer is going in that direction and the `AttitudeEstimator` will sort of expect it.

File: src/fsw/FCCode/AttitudeEstimator.cpp (hunk 2 above)
Comment (kkrol27) [reply src/fsw/FCCode/AttitudeEstimator.cpp:47]: Honestly, for the time being we can just pull magnetometer one or something. With my attitude controller and estimator rewrite I can set up some logic to handle it then.

**Comment by shihaocao** [comment]
I just now added a new section to ADCSCheckoutCase to make sure that magnetometers could function independently, caught a bug, just removed some `return` lines in `imu.cpp`

File: src/fsw/FCCode/ADCSBoxMonitor.cpp (hunk 3 above)
Comment (shihaocao) [reply src/fsw/FCCode/ADCSBoxMonitor.cpp:141]: Done

File: src/fsw/FCCode/AttitudeEstimator.cpp (hunk 2 above)
Comment (shihaocao) [reply src/fsw/FCCode/AttitudeEstimator.cpp]: Yep sounds good

File: src/fsw/FCCode/ADCSBoxMonitor.cpp [hunk 4]
```diff
@@ -115,6 +115,14 @@ bool exceed_bounds(const float input, const float min, const float max){
     return false;
 }
 
+// TODO: WOULD BE A NICE FEATURE IN LIN.hpp
+template<size_t N>
+lin::Vector<float, N> to_linvector(const std::array<float, N>& src) {
+    lin::Vector<float, N> src_cpy;
+    for(unsigned int i = 0; i < 3; i++) src_cpy(i) = src[i];
+    return src_cpy;
+}
```
Comment (kkrol27) [inline src/fsw/FCCode/ADCSBoxMonitor.cpp:124]: This should be marked `static` and the range in the for loop needs to be based on `N`.

**Comment by shihaocao** [comment]
Yep, I re-ran HITL on these new changes. @kkrol27

File: constants [hunk 5]
```diff
@@ -3,35 +3,33 @@ Keep this file in version control so that changes to constants are well-known.
...
+src/adcs/constants.hpp:160: "max_speed_read" = "1047.20f"
+src/adcs/constants.hpp:162: "min_speed_read" = "-max_speed_read"
+src/adcs/constants.hpp:165: "max_speed_command" = "680.678f"
+src/adcs/constants.hpp:167: "min_speed_command" = "-max_speed_command"
+src/adcs/constants.hpp:174: "min_voltage_rd" = "0.0f"
+src/adcs/constants.hpp:176: "max_voltage_rd" = "3.3f"
+src/adcs/constants.hpp:178: "min_voltage_thresh" = "min_voltage_rd"
+src/adcs/constants.hpp:180: "max_voltage_thresh" = "max_voltage_rd"
+src/adcs/constants.hpp:182: "num_sun_sensors" = "20"
+src/adcs/constants.hpp:189: "max_devices" = "32"
```
Comment (shihaocao) [inline constants:32]: No numerical changes were made, just two faulty lines were removed + line numbers changed.

**Pull request merged**